### PR TITLE
GetRelative paths accepts paths shorter than the base directory

### LINKED ISSUE #72043: `GetRelativePath` in `PathUtilities` crashes with IndexOutOfRangeException
**Version Used**: Current develop

I don't know which design limits exist for the `PathUtilities.GetRelativePath(string, string)` method. I saw there is #52936 talking about to replace that method (at least for .NET Core). Both methods however have slightly different behavior currently regarding some cases. 
E.g., .NET returns `.` for same paths, where as Roslyn Utils returns an empty string. However, I don't know whether these differences have an actual impact. 

Still i noticed when using the Roslyn version, in some (trivial) cases the method crashes. Again, idk, what the expectations of this methods are, so maybe these cases are not considered by design.

**Steps to Reproduce**:
Add a test case as follows to FilePathUtilitiesTests.cs
```cs
[Fact]
public void GetRelativePath_CrashingWithTwoUp()
{
    var baseDirectory = @"C:\Alpha\Beta";
    var fullPath = @"C:\";

    var result = PathUtilities.GetRelativePath(baseDirectory, fullPath);

    Assert.Equal(expected: @"..\..", actual: result);
}

[Fact]
public void GetRelativePath_CrashingWithOneUp()
{
    var baseDirectory = @"C:\Alpha";
    var fullPath = @"C:\";

    var result = PathUtilities.GetRelativePath(baseDirectory, fullPath);

    Assert.Equal(expected: @"..", actual: result);
}
```

**Diagnostic Id**:

For each tests the crash has the following stack trace: 

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.

System.IndexOutOfRangeException
Index was outside the bounds of the array.
   at Roslyn.Utilities.PathUtilities.GetRelativePath(String directory, String fullPath) in [redacted]\roslyn\src\Compilers\Core\Portable\FileSystem\PathUtilities.cs:line 567
```

**Expected Behavior**:

The method resolves relative paths correctly. 

**Actual Behavior**:

The method crashes with an `IndexOutOfRangeException`. 

This error occurs if the second parameter has less path parts than the first parameter. 
I think, I already have a fix for that, but i need to test that with some more cases. 
So, if that's an issue that shall be resolve, I can create a PR.

### PR DESCRIPTION
**Pull request by AnakinRaW**
Fixes #72043

### CODE REVIEW

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 1]
```diff
@@ -587,10 +588,22 @@ public static string GetRelativePath(string directory, string fullPath)
...
+                    relativePath = CombinePathsUnchecked(relativePath, fullPathParts[i]);
+                }
+            }
+            else
+            {
+                // There was nothing to add from fullPath, we have a path ending
+                // with either "../" or it is empty. Trim trailing separator in former case.
+                if (!string.IsNullOrEmpty(relativePath))
+                {
+                    relativePath = TrimTrailingSeparators(relativePath);
```
Comment (RikkiGibson) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs:605]: Could you please simplify this by removing the new if/else/if blocks and just calling this method unconditionally after the original loop? It appears to be cheap to call when the relative path does not have trailing separators.

Comment (AnakinRaW) [reply src/Compilers/Core/Portable/FileSystem/PathUtilities.cs]: sure.
done that.

File: src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs [hunk 2]
```diff
@@ -99,5 +99,27 @@ public void GetRelativePath_WithBaseDirectoryMatchingIncompletePortionOfFullPath
 
             Assert.Equal(expected: @"..\Beta2\Gamma", actual: result);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72043")]
+        public void GetRelativePath_CrashingWithTwoUp()
```
Comment (jaredpar) [inline src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs:104]: Need to test the Linux version of this problem in addition to Windows. Recommend making this a `[Theory]` and then use windows and linux paths as the inputs.

File: src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs [hunk 3]
```diff
@@ -99,5 +99,27 @@ public void GetRelativePath_WithBaseDirectoryMatchingIncompletePortionOfFullPath
...
+            var fullPath = @"C:\Gamma";
+
+            var result = PathUtilities.GetRelativePath(baseDirectory, fullPath);
+
+            Assert.Equal(expected: @"..\..\Gamma", actual: result);
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72043")]
+        public void GetRelativePath_CrashingWithOneUp()
+        {
```
Comment (jaredpar) [inline src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs:116]: Same feedback about Linux testing.

**Review by jaredpar — CHANGES_REQUESTED** [review]

File: src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs (hunk 3 above)
Comment (AnakinRaW) [reply src/Workspaces/CoreTest/UtilityTest/FilePathUtilitiesTests.cs:119]: Updated the tests, but made them conditional for Windows and Linux. 
Reason: Just flipping '\' with '/' for the test still returns '\' on Windows systems as this is the primary separator.
On Linux using a `C:\` pattern path feels odd to me and is in fact wrong as `:` is a path separator. I don't want the test to assert this, even the curremt implementation allows '\' and ':' to be used on Linux. But that'd be a different story.

**Review by jaredpar — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Thanks for the contribution @AnakinRaW!

**Pull request merged**